warehouse/query-core/src/test/java/datawave/query/jexl/visitors/LiteralNodeSubsetVisitorTest.java

```diff
@@ -0,0 +1,112 @@
+package datawave.query.jexl.visitors;
+
+import com.google.common.collect.Multimap;
+import datawave.query.jexl.JexlASTHelper;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class LiteralNodeSubsetVisitorTest {
```

Comment:
This whole class could use a refactor. Delegate all tests to a single test method that looks like `test(String query, Map<String,String> expectedLiterals)`.

Then your test methods looks like
```
    @Test
    public void testSingleLiteralWithMatch() throws ParseException {
        String query = "FOO == '1234'";
        Multimap<String,String> literals = Multimaps.newArrayListMultimap();
        literals .put("FOO", "1234");
        test(query, literals );
    }
```

This comment applies to the other test.

Reply:
Done and done.